pkg/driver/controller_server.go

```diff
@@ -94,6 +94,20 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			return nil, status.Errorf(codes.Unavailable, "Volume isn't available to be attached, state is currently %s", v.Status)
 		}
 	}
+
+	// TODO: Uncomment after client implementation is complete.
+	// snapshotID := ""
+	// if volSource := req.GetVolumeContentSource(); volSource != nil {
+	// 	if _, ok := volSource.GetType().(*csi.VolumeContentSource_Snapshot); !ok {
+	// 		return nil, status.Error(codes.InvalidArgument, "Unsupported volumeContentSource type")
+	// 	}
+	// 	snapshot := volSource.GetSnapshot()
+	// 	if snapshot == nil {
+	// 		return nil, status.Error(codes.InvalidArgument, "failed to get snapshot from the volumeContentSource")
```

Comment:
Is this an invalid argument error? This sounds more like an internal error?

Reply:
Thank you for your comment! The intention was to use InvalidArgument, but the error message seems to cause some confusion. `volSource.GetSnapshot() `simply retrieves the content of the protobuf object. Therefore, it returns invalid argument if the value is nil. Perhaps I could consider changing the error message to make it clearer. What do you think? :thinking:

Reply:
The following is an error message example: https://github.com/civo/civo-csi/pull/39/commits/75c436f5448b8c4c32a28c9e8dc6f656b36d38d3

Reply:
Thanks for the great clarity!